mos-platform/nes-gtrom/_prg-rom-banked.ld

```diff
@@ -0,0 +1,252 @@
+__prg_rom_10_offset = 0x050000;
+__prg_rom_11_offset = 0x058000;
+__prg_rom_12_offset = 0x060000;
+__prg_rom_13_offset = 0x068000;
+__prg_rom_14_offset = 0x070000;
+__prg_rom_15_offset = 0x078000;
+
+MEMORY {
+  /* PRG-ROM LMAs. */
+  prg_rom_0 : ORIGIN = __prg_rom_0_lma, LENGTH = __prg_rom_size >= 32 ? 0x8000 - 0x6 : 0
+  prg_rom_1 : ORIGIN = __prg_rom_1_lma, LENGTH = __prg_rom_size >= 64 ? 0x8000 - 0x6 : 0
+  prg_rom_2 : ORIGIN = __prg_rom_2_lma, LENGTH = __prg_rom_size >= 128 ? 0x8000 - 0x6 : 0
+  prg_rom_3 : ORIGIN = __prg_rom_3_lma, LENGTH = __prg_rom_size >= 128 ? 0x8000 - 0x6 : 0
+  prg_rom_4 : ORIGIN = __prg_rom_4_lma, LENGTH = __prg_rom_size >= 256 ? 0x8000 - 0x6 : 0
+  prg_rom_5 : ORIGIN = __prg_rom_5_lma, LENGTH = __prg_rom_size >= 256 ? 0x8000 - 0x6 : 0
+  prg_rom_6 : ORIGIN = __prg_rom_6_lma, LENGTH = __prg_rom_size >= 256 ? 0x8000 - 0x6 : 0
+  prg_rom_7 : ORIGIN = __prg_rom_7_lma, LENGTH = __prg_rom_size >= 256 ? 0x8000 - 0x6 : 0
+  prg_rom_8 : ORIGIN = __prg_rom_8_lma, LENGTH = __prg_rom_size >= 512 ? 0x8000 - 0x6 : 0
+  prg_rom_9 : ORIGIN = __prg_rom_9_lma, LENGTH = __prg_rom_size >= 512 ? 0x8000 - 0x6 : 0
+  prg_rom_10 : ORIGIN = __prg_rom_10_lma, LENGTH = __prg_rom_size >= 512 ? 0x8000 - 0x6 : 0
+  prg_rom_11 : ORIGIN = __prg_rom_11_lma, LENGTH = __prg_rom_size >= 512 ? 0x8000 - 0x6 : 0
+  prg_rom_12 : ORIGIN = __prg_rom_12_lma, LENGTH = __prg_rom_size >= 512 ? 0x8000 - 0x6 : 0
+  prg_rom_13 : ORIGIN = __prg_rom_13_lma, LENGTH = __prg_rom_size >= 512 ? 0x8000 - 0x6 : 0
+  prg_rom_14 : ORIGIN = __prg_rom_14_lma, LENGTH = __prg_rom_size >= 512 ? 0x8000 - 0x6 : 0
+  prg_rom_15 : ORIGIN = __prg_rom_15_lma, LENGTH = __prg_rom_size >= 512 ? 0x8000 - 0x6 : 0
+  vectors : ORIGIN = 0x10000 - 0x6, LENGTH = 6
+
+  /* CHR-ROM LMAs. */
+  chr_rom_0   : ORIGIN = 0x01000000, LENGTH = __chr_rom_size >= 8    ? 0x2000 : 0
+  chr_rom_1   : ORIGIN = 0x01002000, LENGTH = __chr_rom_size >= 16   ? 0x2000 : 0
+}
+
+/**
+ * Alias "c_readonly" to prg_rom_0. Everything placed in this memory region
+ * before the .prg_rom_fixed section is defined below (in file parsing order)
+ * will become part of the fixed bank.
+ *
+ * So long as all the read-only C sections have been allocated to the
+ * "c_readonly" region before this happens, the invariant of keeping all
+ * default C section code in a fixed section is maintained.
+ */
+REGION_ALIAS("c_readonly", prg_rom_0)
+
+SECTIONS {
+  /* Define a fixed section at the beginning of PRG-ROM bank 0. */
+  .prg_rom_fixed : {
+    *(.prg_rom_fixed .prg_rom_fixed.*)
+    __prg_rom_fixed_end = .;
+    __prg_rom_fixed_size = __prg_rom_fixed_end - __prg_rom_0_lma;
+  } >prg_rom_0
+
+  /* Offset each non-fixed bank section's LMA by the fixed section. */
+  /* This is accounted for in the custom output format. */
+  .prg_rom_0 __prg_rom_0_lma + __prg_rom_fixed_size : {
+      /* Non-fixed bank */
+      *(.prg_rom_0 .prg_rom_0.*)
+      
+      /* Non-fixed DPCM */
+      . = ABSOLUTE(.) < 0xc000 ? 0xc000 : ALIGN(64);
```

Comment:
Ah, similarly, there was recently cleanup work done to `dpcm.ld` to break out `.dpcm` into its own section. Right now, any filler inserted by the set to `.` will be accounted for in the linker map to `.prg_rom_0`, obscuring how much was actually used. Similarly, such sections can be defined as `.dpcm_k (ABSOLUTE(.) < 0xc000 ? 0xc000 : .) : ALIGN(64) {}`

Reply:
Sorry, this should be `.dpcm_k (ABSOLUTE(.) < 0xc000 ? 0xc000 : ALIGN(64)) {}`; it doesn't automatically align an explicit address.

Reply:
It turns out they needed a bit more fixing than that, but hey, there's tests now!